pubsub/subscriptions/subscription_test.go

```diff
@@ -155,6 +158,103 @@ func TestIAM(t *testing.T) {
 	})
 }
 
+func TestPullMsgsSync(t *testing.T) {
+	ctx := context.Background()
+	tc := testutil.SystemTest(t)
+	client := setup(t)
+
+	topic := client.Topic(topicName)
+	ok, err := topic.Exists(ctx)
+	if err != nil {
+		t.Fatalf("failed to check if topic exists: %v", err)
+	}
+	if !ok {
+		topic, err := client.CreateTopic(ctx, topicName)
+		if err != nil {
+			t.Fatalf("CreateTopic: %v", err)
+		}
+		_, err = client.CreateSubscription(ctx, subName, pubsub.SubscriptionConfig{
+			Topic:       topic,
+			AckDeadline: 20 * time.Second,
+		})
+		if err != nil {
+			t.Fatalf("CreateSubscription: %v", err)
+		}
+	}
+
+	// Publish 10 messages on the topic.
+	var results []*pubsub.PublishResult
+	for i := 0; i < 10; i++ {
+		res := topic.Publish(ctx, &pubsub.Message{
+			Data: []byte(fmt.Sprintf("hello world #%d", i)),
+		})
+		results = append(results, res)
+	}
+
+	// Check that all messages were published.
+	for _, r := range results {
+		if _, err := r.Get(ctx); err != nil {
+			t.Fatalf("Get publish result: %v", err)
+		}
+	}
+
+	maxMessages := 5
+	buf := new(bytes.Buffer)
+	err = pullMsgsSync(buf, tc.ProjectID, subName, int32(maxMessages))
+	if err != nil {
+		t.Fatalf("failed to pull messages: %v", err)
+	}
+	got := buf.String()
+	want := "Got message \"hello world #4\""
+	if !strings.Contains(got, want) {
+		t.Fatalf("got output: %v\nwant %v", got, want)
+	}
```

Comment:
There's some redundancy in the test. Ten messages got published, but the script pulls for 5 of them, then checks if message #4 is there. Can we pull all ten messages and make sure message #0-9 are all there?

Reply:
I want the sync pull test to only pull max 5 messages. The publish methods sends 10 so that we can test that sync pull works properly. I can change the test to not be 0 indexed to be less confusing if that helps?

Reply:
Okay, publishing more than the number you want to pull makes sense. 

I think Indexed messages are fine. I guess I don't understand how you could make sure `#4` is in among the pulled messages.

Reply:
Oh you're totally right. Maybe I'll check for number of lines instead then. Good catch!